packages/zowe-explorer-api/src/utils/DeferredPromise.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+/* Status of the deferred promise */
+export type DeferredPromiseStatus = "pending" | "fulfilled" | "rejected";
```

Comment:
Should this be an enum instead of a string union type?

Reply:
Either would realistically work fine, I think its just a matter of preference - but to make comparisons easier, I'll change it to an enum.

Resolved in fd55a0f